File: color/src/main/java/com/tobiasschuerg/color/ColorCreator.java
package com.tobiasschuerg.color;

import com.tobiasschuerg.color.random.RandomColor;

/**
 * Created by Tobias
 */
public class ColorCreator {

    /*
    Returns a random color
     */
    public static int getRandomColor() {
        return getRandomColors(1)[0];
    }

    /**
     * Returns an array of x random color values
     */
    public static int[] getRandomColors(int x) {
        RandomColor randomColor = new RandomColor();
        return randomColor.randomColor(x);
    }


    /*
       Returns an array of ten green colors
     */
    public static int[] getSimilarRandomColors(RandomColor.Color color, int count) {
        RandomColor randomColor = new RandomColor();
        return randomColor.random(color, count);
    }
}

File: color/src/main/java/com/tobiasschuerg/color/ColorCreator.kt
package com.tobiasschuerg.color

import com.tobiasschuerg.color.models.HSVColor
import com.tobiasschuerg.color.random.Color
import com.tobiasschuerg.color.random.RandomColor

/**
 * Created by Tobias
 */
object ColorCreator {
    /*
    Returns a random color
     */
    fun randomColor() = RandomColor.randomColor()

    /**
     * Returns an array of count random color values
     */
    fun getRandomColors(count: Int): List<HSVColor> = RandomColor.randomColors(count)

    /*
       Returns an array of ten green colors
     */
    internal fun getSimilarRandomColors(
        color: Color,
        count: Int,
    ): List<HSVColor> {
        return RandomColor.random(color, count)
    }
}
